### PR TITLE
fix: drop rusha

### PR DESCRIPTION
**Pull request by ThaUnknown**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
replaced simple-sha1 with uint8-util, thus dropping rusha, replaced all buffer with uint8array

main function is now async, so this is breaking
**Which issue (if any) does this pull request address?**
https://github.com/webtorrent/webtorrent/pull/2390
**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: test/corrupt.js [hunk 1]
```diff
@@ -2,9 +2,11 @@ import fixtures from 'webtorrent-fixtures'
...
 
-test('exception thrown when torrent file is missing `name` field', t => {
-  t.throws(() => {
-    parseTorrent(fixtures.corrupt.torrent)
-  })
+test('exception thrown when torrent file is missing `name` field', async t => {
+  try {
+    await parseTorrent(fixtures.corrupt.torrent)
+  } catch (e) {
+    t.ok(e instanceof Error)
```
Comment (DiegoRBaquero) [inline test/corrupt.js:9]: shouldn't there be a t.error after the await to fail the test?

File: test/non-torrent-magnet.js [hunk 2]
```diff
@@ -13,9 +13,13 @@ test('exception thrown with non-bittorrent URNs', function (t) {
...
   ]
 
-  magnets.forEach(function (magnet) {
-    t.throws(() => parseTorrent(magnet))
-  })
+  await Promise.all(magnets.map(async function (magnet) {
+    try {
+      await parseTorrent(magnet)
+    } catch (e) {
+      t.ok(e instanceof Error)
```
Comment (DiegoRBaquero) [inline test/non-torrent-magnet.js:20]: same here

File: test/corrupt.js (hunk 1 above)
Comment (ThaUnknown) [reply test/corrupt.js:9]: I see, yeah I'll add that

**Review by DiegoRBaquero — APPROVED** [review]

**Comment by klchoi** [comment]
@ThaUnknown Sorry for bumping an old PR, but would you help me understand the decision of the breaking change?

From the diff, looks like `parseTorrent` become async is because `decodeTorrentFile` is async which is because of `hash` function from [uint8-util](https://github.com/ThaUnknown/uint8-util/blob/83b6752377c536188ad96e7abe3594fff572f703/_node.js#L23-L27) is async. But since we only operate on the buffer, no I/O, no networking, then making it async doesn't seem to solve any problem.

**Comment by ThaUnknown** [comment]
to make it faster in browser

**Comment by ThaUnknown** [comment]
@klchoi see https://github.com/ThaUnknown/uint8-util/blob/83b6752377c536188ad96e7abe3594fff572f703/browser.js#L56-L61

**Comment by klchoi** [comment]
@ThaUnknown thanks for clarifying. I guess the native apis make more sense in webtorrent as we're hashing huge files there, but for this package we only hash torrent files which is really tiny. It's sad we're loosing the beautiful sync api.

Sorry for the rant. I really like the sync interface in nodejs.

Thanks for your contribution.

**Pull request merged**